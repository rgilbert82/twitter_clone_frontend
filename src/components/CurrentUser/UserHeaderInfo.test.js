import React from 'react';
import { UserHeaderInfo } from '.';
import { shallow } from 'enzyme';

describe(UserHeaderInfo, () => {
  const currentUserData = { id: 1, name: 'Ryan', username: 'RG' };
  let component;

  beforeEach(() => {
    component = shallow(<UserHeaderInfo appLogout={jest.fn()} currentUser={ currentUserData }/>)
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });
});
