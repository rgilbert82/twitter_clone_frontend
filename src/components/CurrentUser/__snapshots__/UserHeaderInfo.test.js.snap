// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class UserHeaderInfo extends _react2.default.Component {
  render() {
    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this },
        _react2.default.createElement(_reactBootstrap.Nav, { pullLeft: true, __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this },
          _react2.default.createElement(_reactBootstrap.NavDropdown, { eventKey: 1, title: 'Options', id: 'basic-nav-dropdown', __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this },
            _react2.default.createElement(_reactBootstrap.MenuItem, { __source: { fileName: _jsxFileName, lineNumber: 12 }, __self: this }, _react2.default.createElement(_reactRouterDom.Link, { to: '/account', __source: { fileName: _jsxFileName, lineNumber: 12 }, __self: this }, 'Edit Account')))),


        _react2.default.createElement(_reactBootstrap.Nav, { pullRight: true, className: 'header_session_options', __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this },
          _react2.default.createElement(_Sessions.LogoutForm, { appLogout: this.props.appLogout, __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }))));



  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserHeaderInfo
    appLogout={[Function]}
    currentUser={
        Object {
            "id": 1,
            "name": "Ryan",
            "username": "RG",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Nav
          bsClass="nav"
          justified={false}
          pullLeft={true}
          pullRight={false}
          stacked={false}
>
          <NavDropdown
                    eventKey={1}
                    id="basic-nav-dropdown"
                    title="Options"
          >
                    <MenuItem
                              bsClass="dropdown"
                              disabled={false}
                              divider={false}
                              header={false}
                    >
                              <Link
                                        replace={false}
                                        to="/account"
                              >
                                        Edit Account
                              </Link>
                    </MenuItem>
          </NavDropdown>
</Nav>,
        <Nav
          bsClass="nav"
          className="header_session_options"
          justified={false}
          pullLeft={false}
          pullRight={true}
          stacked={false}
>
          <LogoutForm
                    appLogout={[Function]}
          />
</Nav>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "nav",
          "children": <NavDropdown
            eventKey={1}
            id="basic-nav-dropdown"
            title="Options"
>
            <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={false}
            >
                        <Link
                                    replace={false}
                                    to="/account"
                        >
                                    Edit Account
                        </Link>
            </MenuItem>
</NavDropdown>,
          "justified": false,
          "pullLeft": true,
          "pullRight": false,
          "stacked": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              header={false}
>
              <Link
                            replace={false}
                            to="/account"
              >
                            Edit Account
              </Link>
</MenuItem>,
            "eventKey": 1,
            "id": "basic-nav-dropdown",
            "title": "Options",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "dropdown",
              "children": <Link
                replace={false}
                to="/account"
>
                Edit Account
</Link>,
              "disabled": false,
              "divider": false,
              "header": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Edit Account",
                "replace": false,
                "to": "/account",
              },
              "ref": null,
              "rendered": "Edit Account",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "nav",
          "children": <LogoutForm
            appLogout={[Function]}
/>,
          "className": "header_session_options",
          "justified": false,
          "pullLeft": false,
          "pullRight": true,
          "stacked": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "appLogout": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Nav
            bsClass="nav"
            justified={false}
            pullLeft={true}
            pullRight={false}
            stacked={false}
>
            <NavDropdown
                        eventKey={1}
                        id="basic-nav-dropdown"
                        title="Options"
            >
                        <MenuItem
                                    bsClass="dropdown"
                                    disabled={false}
                                    divider={false}
                                    header={false}
                        >
                                    <Link
                                                replace={false}
                                                to="/account"
                                    >
                                                Edit Account
                                    </Link>
                        </MenuItem>
            </NavDropdown>
</Nav>,
          <Nav
            bsClass="nav"
            className="header_session_options"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
>
            <LogoutForm
                        appLogout={[Function]}
            />
</Nav>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "nav",
            "children": <NavDropdown
              eventKey={1}
              id="basic-nav-dropdown"
              title="Options"
>
              <MenuItem
                            bsClass="dropdown"
                            disabled={false}
                            divider={false}
                            header={false}
              >
                            <Link
                                          replace={false}
                                          to="/account"
                            >
                                          Edit Account
                            </Link>
              </MenuItem>
</NavDropdown>,
            "justified": false,
            "pullLeft": true,
            "pullRight": false,
            "stacked": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MenuItem
                bsClass="dropdown"
                disabled={false}
                divider={false}
                header={false}
>
                <Link
                                replace={false}
                                to="/account"
                >
                                Edit Account
                </Link>
</MenuItem>,
              "eventKey": 1,
              "id": "basic-nav-dropdown",
              "title": "Options",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "dropdown",
                "children": <Link
                  replace={false}
                  to="/account"
>
                  Edit Account
</Link>,
                "disabled": false,
                "divider": false,
                "header": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Edit Account",
                  "replace": false,
                  "to": "/account",
                },
                "ref": null,
                "rendered": "Edit Account",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "nav",
            "children": <LogoutForm
              appLogout={[Function]}
/>,
            "className": "header_session_options",
            "justified": false,
            "pullLeft": false,
            "pullRight": true,
            "stacked": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appLogout": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
